src/components/application_manager/test/commands/mobile/alert_maneuver_request_test.cc

```diff
@@ -172,18 +182,141 @@ TEST_F(AlertManeuverRequestTest, Run_ProcessingResult_SUCCESS) {
+
+  EXPECT_CALL(mock_message_helper_, SubscribeApplicationToSoftButton(_, _, _));
+
+  EXPECT_CALL(mock_app_manager_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::Navigation_AlertManeuver)));
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::TTS_Speak)));
+  command->Run();
+}
+
 TEST_F(AlertManeuverRequestTest, OnEvent_ReceivedUnknownEvent_UNSUCCESS) {
   CommandPtr command(CreateCommand<AlertManeuverRequest>());
   Event event(hmi_apis::FunctionID::INVALID_ENUM);
-
   MessageSharedPtr result_msg(
       CatchMobileCommandResult(CallOnEvent(*command, event)));
+
   EXPECT_EQ(mobile_apis::Result::INVALID_ENUM,
             static_cast<mobile_apis::Result::eType>(
                 (*result_msg)[am::strings::msg_params][am::strings::result_code]
                     .asInt()));
 }
 
+TEST_F(AlertManeuverRequestTest,
+       OnEvent_ReceivedNavigationAlertManeuverEvent_SUCCESS) {
+  Event event(hmi_apis::FunctionID::Navigation_AlertManeuver);
+
+  hmi_apis::Common_Result::eType response_code =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg_)[am::strings::params][am::hmi_response::code] = response_code;
+  mobile_apis::Result::eType casted_response_code =
+      static_cast<mobile_apis::Result::eType>(response_code);
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg_));
+
+  EXPECT_CALL(mock_message_helper_, MobileToHMIResult(casted_response_code))
+      .WillOnce(Return(hmi_apis::Common_Result::SUCCESS));
+  event.set_smart_object(*msg_);
+  command->on_event(event);
+}
+
+TEST_F(AlertManeuverRequestTest,
+       OnEvent_ReceivedNavigationAlertManeuverEvent_UNSUCCESS) {
+  Event event(hmi_apis::FunctionID::Navigation_AlertManeuver);
+
+  hmi_apis::Common_Result::eType response_code =
+      hmi_apis::Common_Result::GENERIC_ERROR;
+  (*msg_)[am::strings::params][am::hmi_response::code] = response_code;
+  mobile_apis::Result::eType casted_response_code =
+      static_cast<mobile_apis::Result::eType>(response_code);
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg_));
+
+  EXPECT_CALL(mock_message_helper_, MobileToHMIResult(casted_response_code))
+      .WillOnce(Return(hmi_apis::Common_Result::GENERIC_ERROR));
+  event.set_smart_object(*msg_);
+  command->on_event(event);
+}
+
+TEST_F(AlertManeuverRequestTest, OnEvent_ReceivedTTSSpeakEvent_SUCCESS) {
+  Event event(hmi_apis::FunctionID::TTS_Speak);
+
+  hmi_apis::Common_Result::eType response_code =
+      hmi_apis::Common_Result::GENERIC_ERROR;
+  (*msg_)[am::strings::params][am::hmi_response::code] = response_code;
+  mobile_apis::Result::eType casted_response_code =
+      static_cast<mobile_apis::Result::eType>(response_code);
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg_));
+
+  EXPECT_CALL(mock_message_helper_, MobileToHMIResult(casted_response_code))
+      .WillOnce(Return(hmi_apis::Common_Result::GENERIC_ERROR));
```

Comment:
@indlu, Do we return `hmi_apis::Common_Result::GENERIC_ERROR` and expect success, but while return `hmi_apis::Common_Result::SUCCESS` expect unsuccess? 

This looks strange.